simple

```diff
@@ -220,16 +232,46 @@ docker)
 
 ###############################################################################
 validate)
+    (
+	[ -e $cli_conf_dir ] || (
+            mkdir $cli_conf_dir
+        ) || exit
+
+        [ -e $installed_pip ] || (
+            yum install -y python-pip && pip install --upgrade pip &&
+            touch $installed_pip
+        ) || exit
+
+        [ -e $installed_virtualenv ] || (
+            yum install -y python-virtualenv && touch $installed_virtualenv
+        ) || exit
+
+	[ -e $installed_git ] || (
+	    yum install -y git && touch $installed_git
+	) || exit
+	echo $validator_dir
+        [ -e $installed_validator ] || (
+            mkdir -p $validator_dir &&  cd $validator_dir && pwd &&
+	    git clone "$validator_url" $validator_dir &&
+	    virtualenv $validator_env && source $validator_env/bin/activate &&
+	    pip install -r requirements.txt && pip install $validator_dir &&
```

Comment:
is the argument to the second pip install correct?

Reply:
Yup, the first one installs the dependencies while the second one installs the validation engine itself.

Reply:
The 2nd pip install has a _directory_ as an argument, instead of a product...

Reply:
Yup, that is intentional. Pip can install packages from source code as well. The directory contains the product i.e. the source of the infra validation engine python package which is not available on PyPi as I have not released it. Therefore, I clone the code from GitHub and install the package with pip as seen above.